src/scripts.js

```diff
@@ -162,42 +164,43 @@ function displayRecipeDetailPage(event) {
     else {
         hide([removeRecipeButton])
     }
-    show([favoriteButton])
-    hide([cookRecipeButton])
     displayRecipeInstructions(event)
     displayRecipeTotalCost(event)
     displayRecipeIngredients(event)
     if(user.recipesToCook.includes(foundRecipe)) {
         hide([favoriteRecipeButton])
         show([cookRecipeButton])
     } 
-    hide([ingredientsNeededToCook])
-    console.log('pantry', user.pantry)
+    hide([ingredientsNeededToCook, userCanCook, cookRecipeButton])
+    show([cookStatusSection, ingredientsNeededToCook, favoriteButton])
     user.pantry.checkPantryForIngredients(foundRecipe)
     user.pantry.determineIngredientsNeeded(foundRecipe)
-    missingIngredientList.innerHTML = ''
-    if(user.recipesToCook.includes(foundRecipe) && user.pantry.userCanCook) {
-        show([cookStatusSection])
-    } else if (user.recipesToCook.includes(foundRecipe) && !user.pantry.userCanCook) {
-        console.log("HERE", user.pantry.ingredientsNeeded)
-        const displayThese = user.pantry.ingredientsNeeded.map((ingredientNeed)=>{
-           let ingredientName = ingredients.ingredients.reduce((name, ingredient)=>{
-               if (ingredientNeed.missingIngredient === ingredient.id) {
-                name = ingredient.name
-               }
-               return name
-            }, "")
-            return {name: ingredientName, quantity: ingredientNeed.quantityNeeded, unit:ingredientNeed.units} 
-        })
-        displayThese.forEach((missing)=>{
-            missingIngredientList.innerHTML += `<li>${missing.quantity} ${missing.unit} ${missing.name}</li>`
-        })
-        show([cookStatusSection])
-        hide([userCanCook])
+    if(user.pantry.userCanCook) {
+        show([cookRecipeButton])
+        hide([ingredientsNeededToCook])
+    } else {
+        displayMissingIngr()
         show([ingredientsNeededToCook])
     }
 }
 
+function displayMissingIngr() {
+    missingIngredientList.innerHTML = ''
+    // user.pantry.checkPantryForIngredients(foundRecipe)
+    // user.pantry.determineIngredientsNeeded(foundRecipe)
+    return user.pantry.ingredientsNeeded.map((ingredientNeed)=>{
+        let ingredientName = ingredients.ingredients.reduce((name, ingredient)=>{
+            if (ingredientNeed.missingIngredient === ingredient.id) {
+             name = ingredient.name
+            }
+            return name
+         }, "")
+         return {name: ingredientName, quantity: ingredientNeed.quantityNeeded, unit:ingredientNeed.units} 
+     }).forEach((missing)=>{
+        missingIngredientList.innerHTML += `<li>${missing.quantity} ${missing.unit} ${missing.name}</li>`
```

Comment:
could we format the missing.quantity with toFixed(2). When a recipe is missing 1/3 cup of something, it displays it as 0.33333333333333333333333

Reply:
That's a great catch Kiko! I think that we also should add `.toLowerCase()` to the units, the data is inconsistent and sometimes the unit is in caps.